src/core/symbolic.cc

```diff
@@ -10,7 +10,7 @@
 namespace nnvm {
 
 namespace symbol_constants {
-const char *kNamespaceSeparator = "_";
+const char *kNamespaceSeparator = "$";
```

Comment:
saved json will be upgraded
the only thing that's broken is list_attr(recursive=True)
I raised a warning when it's called and replaced it with attr_dict